test/libc/sock/unix_test.c

```diff
@@ -70,17 +70,20 @@ TEST(unix, datagram) {
   munmap(ready, 1);
 }
 
-void StreamServer(atomic_bool *ready) {
+void StreamServer(atomic_bool *ready, const char *path, bool check_path) {
   char buf[256] = {0};
   uint32_t len = sizeof(struct sockaddr_un);
-  struct sockaddr_un addr = {AF_UNIX, "foo.sock"};
+  struct sockaddr_un addr = {AF_UNIX};
+  strcpy(addr.sun_path, path);
   ASSERT_SYS(0, 3, socket(AF_UNIX, SOCK_STREAM, 0));
+  unlink(path);
+  errno = 0;
   ASSERT_SYS(0, 0, bind(3, (void *)&addr, len));
   bzero(&addr, sizeof(addr));
   ASSERT_SYS(0, 0, getsockname(3, (void *)&addr, &len));
-  ASSERT_EQ(2 + 8 + 1, len);
+  if (check_path) ASSERT_EQ(2 + strlen(path) + 1, len);
```

Comment:
Assuming `getsockname` and `getpeername` is implemented for AF_UNIX on Windows, implementing them for Windows AF_UNIX in Cosmo would be greatly appreciated for both the functionality and to avoid making tests platform specific as much as possible.

Reply:
the problem is the path is no longer starts with /tmp (because it has been rewritten). and there is no way to recover it(unable to distinguish whether it is a real path manually set by the user)

Reply:
Sounds like there's no easy way to implement these as Linux does, we can skip these for now. I noticed in various situations on Linux and FreeBSD when unimplemented return a `sun_path` of `\0` and out length of `3`.